frontend/js/src/settings/notification-settings/NotificationSettings.tsx

```diff
@@ -0,0 +1,163 @@
+    e.preventDefault();
+    setSaving(true);
+    try {
+      const response = await fetch("/settings/digest-setting/", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          digest: digestEnabled && notificationsEnabled,
+          digest_age: digestAge,
+        }),
+      });
+      if (!response.ok) {
+        throw new Error(`${response.status} HTTP response.`);
+      }
+      toast.success("Notification settings saved successfully");
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Could not update notification settings.", error);
+      toast.error("Failed to save notification settings. Please try again.");
+    } finally {
+      setSaving(false);
+    }
+  };
+
+  const hasNoChanges =
+    digestEnabled === initialDigestEnabled && digestAge === initialDigestAge;
+
+  if (loading) {
+    return <div>Loading notification settings...</div>;
+  }
+
+  return (
+    <>
+      <Helmet>
+        <title>Notification Settings</title>
+      </Helmet>
+      <h2 className="page-title">Notification settings</h2>
+      <p>
+        We will always email you right away about{" "}
+        <b>important account issues</b> (like problems with connected services).
+        <br />
+        For everything else, you can choose to turn off notifications or receive
+        them in a periodic digest.
+      </p>
+      <form onSubmit={updateDigestSettings}>
+        <Switch
+          id="enable-notifications"
+          checked={notificationsEnabled}
+          onChange={(e) => setNotificationsEnabled(!notificationsEnabled)}
+          value="notifications"
+          switchLabel={
+            <span
+              className={`text-brand ${
+                !notificationsEnabled ? "text-muted" : ""
+              }`}
+            >
+              <span>Enable e-mail notifications</span>
+            </span>
+          }
+        />
+        <details open={notificationsEnabled}>
+          <summary
+            className="mt-4"
+            onClick={(e) => {
+              e.preventDefault();
+            }}
+          >
+            {!notificationsEnabled && (
+              <p className="alert alert-info mb-0 d-inline-block">
+                You will still receive important e-mails that relate to your
+                account
+              </p>
+            )}
+          </summary>
+          <h3 className="mt-4">Digest options</h3>
+          <p>Receive your notifications grouped into an e-mail</p>
+          <Switch
+            id="enable-digest"
+            value="digest"
+            checked={digestEnabled}
+            onChange={(e) => setDigestEnabled(!digestEnabled)}
+            switchLabel={
+              <span className={`h4 ${!digestEnabled ? "text-muted" : ""}`}>
+                <span>Enable digest</span>
+              </span>
+            }
```

Comment:
You could use utility CSS classes (`ms-2` and `me-2` for example, numbers to be chosen appropriately) to add a margin before and after instead of using spaces.